ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/CustomObjectField.js

```diff
@@ -0,0 +1,328 @@
+    const {
+      schema : rawSchema,
+      uiSchema = {},
+      formData,
+      errorSchema,
+      idSchema,
+      name,
+      required = false,
+      disabled = false,
+      readonly = false,
+      hideError,
+      idPrefix,
+      idSeparator,
+      onBlur,
+      onFocus,
+      registry,
+    } = this.props;
+
+    const { fields, formContext, schemaUtils } = registry;
+    const { SchemaField } = fields;
+    const schema = schemaUtils.retrieveSchema(rawSchema, formData);
+    const uiOptions = getUiOptions(uiSchema);
+    const { properties : schemaProperties = {} } = schema;
+
+    const title = schema.title === undefined ? name : schema.title;
+    const description = uiOptions.description || schema.description;
+    let orderedProperties;
+    try {
+      const properties = Object.keys(schemaProperties);
+      orderedProperties = orderProperties(properties, uiOptions.order);
+    } catch (err) {
+      return (
+        <div>
+          <p className="config-error" style={{ color : "red" }}>
+            Invalid {name || "root"} object field configuration:
+            <em>{(err).message}</em>.
+          </p>
+          <pre>{JSON.stringify(schema)}</pre>
+        </div>
+      );
+    }
+
+    const Template = getTemplate(
+      "ObjectFieldTemplate",
+      registry,
+      uiOptions
+    );
+
+    const templateProps = {
+      title : uiOptions.title || title,
+      description,
+      rawErrors : this.props.rawErrors,
+      properties : orderedProperties.map((name) => {
+        const addedByAdditionalProperties = has(schema, [
+          PROPERTIES_KEY,
+          name,
+          ADDITIONAL_PROPERTY_FLAG,
+        ]);
+        const fieldUiSchema = addedByAdditionalProperties
+          ? uiSchema.additionalProperties
+          : uiSchema[name];
+        const hidden = getUiOptions(fieldUiSchema).widget === "hidden";
+        const fieldIdSchema = get(idSchema, [name], {});
+
+        return {
+          content : (
+            <SchemaField
+              key={name}
+              name={name}
+              required={this.isRequired(name)}
+              schema={get(schema, [PROPERTIES_KEY, name], {})}
+              uiSchema={fieldUiSchema}
+              errorSchema={get(errorSchema, name)}
+              idSchema={fieldIdSchema}
+              idPrefix={idPrefix}
+              idSeparator={idSeparator}
+              formData={get(formData, name)}
+              formContext={formContext}
+              wasPropertyKeyModified={this.state.wasPropertyKeyModified}
+              onKeyChange={this.onKeyChange(name)}
+              onChange={this.onPropertyChange(
+                name,
+                addedByAdditionalProperties
+              )}
+              onBlur={onBlur}
+              onFocus={onFocus}
+              registry={registry}
+              disabled={disabled}
+              readonly={readonly}
+              hideError={hideError}
+              onDropPropertyClick={this.onDropPropertyClick}
```

Comment:
This is a custom field-specific to the Meshery project?

Reply:
A bit confusing, I agree.